tests/TestDeferredTestReporter.cpp

```diff
@@ -54,7 +54,7 @@ namespace UnitTest
       {
          reporter.ReportTestStart(details);
 
-         DeferredTestResult const& result = reporter.GetResults().at(0);
+         DeferredTestResult const& result = reporter.GetResults()[0];
```

Comment:
I don't want to lose the exception-throwing behavior of .at in these tests. If this is truly necessary, then we may need another way.